### PR TITLE
command to add new member

### PR DESCRIPTION
**Pull request by abidhkm**
Command for adding a new member to the member list.

### CODE REVIEW

File: cmd/add-member.cmd.ts [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    if (!fs.existsSync(directory)) {
+      fs.mkdirSync(directory);
+      return true;
+    } else {
+      return "username already exist";
+    }
+  }
+}
+
+async function addMember() {
```
Comment (nilbuild) [inline cmd/add-member.cmd.ts]: No need for this separate function, let's put the logic inside the same command body

File: cmd/add-member.cmd.ts [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        message: "What is your company name?",
+        validate: validateMinLength,
+      },
+      {
+        name: "imageUrl",
+        type: "input",
+        message: "What is your imageUrl?",
+      },
+    ])
+    .then(async (answer: Answer) => {
```
Comment (nilbuild) [inline cmd/add-member.cmd.ts]: Don't mix `await` and `.then`, we can just do

```typescript
const answer: Answer = await inquirer.prompt(..)
```

File: cmd/add-member.cmd.ts [hunk 3]
```diff
@@ -0,0 +1,125 @@
+import fs from "fs"
+import fetch from "node-fetch"
+import fileType from "file-type"
+import inquirer from "inquirer"
+import { Command } from "./index";
+
+const IMAGE_DIR = "./public/avatars";
+const MEMBERS_DIR = "./data/members";
```
Comment (nilbuild) [inline cmd/add-member.cmd.ts]: Use camel case for the constants and also for the values use `process.cwd()`

```typescript
const avatarsDir = path.join(process.cwd(), './public/avatars');
const membersDir = path.join(process.cwd(), './public/members');
```

File: cmd/add-member.cmd.ts [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+async function downloadAndSaveImg(answer: Answer) {
+  try {
+    const response = await fetch(answer.imageUrl);
+    const buffer = await response.buffer();
+    const fileDetails = await fileType.fromBuffer(buffer);
+    const fileName = `${answer.username}.${fileDetails.ext}`;
+
+    fs.writeFile(`${IMAGE_DIR}/${fileName}`, buffer, () => {
+      console.log("Image saved!");
+    });
```
Comment (nilbuild) [inline cmd/add-member.cmd.ts]: Just use `fs.writeFileSync(path.join(avatarsDir, fileName), buffer)` no need for callback

File: cmd/add-member.cmd.ts [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+
+    return `/avatars/${fileName}`;
+  } catch (e) {
+    console.log("Something went wrong, couldn't save image;", e);
+  }
```
Comment (nilbuild) [inline cmd/add-member.cmd.ts]: Make the function just return the `fileName` or `false` and put the check in the caller and exit with error if the image couldn't be saved.

File: cmd/add-member.cmd.ts [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+        validate: validateMinLength,
+      },
+      {
+        name: "imageUrl",
+        type: "input",
+        message: "What is your imageUrl?",
+      },
+    ])
+    .then(async (answer: Answer) => {
+      const directory = `${MEMBERS_DIR}/${answer.username}`;
```
Comment (nilbuild) [inline cmd/add-member.cmd.ts]: Always use `path.join` when joining paths
```suggestion
      const memberDir = path.join(membersDir, answer.username);
```

File: cmd/add-member.cmd.ts [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+      };
+
+      // profile
+      fs.writeFile(
+        `${directory}/profile.json`,
+        JSON.stringify(userDetails),
+        function (err: Error) {
+          if (err) throw err;
+        }
+      );
```
Comment (nilbuild) [inline cmd/add-member.cmd.ts]: Just use `fs.writeFileSync`, use `path.join` to combine directory path with the filename and format the JSON

```typescript
fs.writeFileSync(memberProfilePath, JSON.stringify(null, true, 2))
```

File: cmd/add-member.cmd.ts [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+
+      // bio.md
+      fs.writeFile(`${directory}/bio.md`, "", function (err: Error) {
+        if (err) throw err;
+      });
```
Comment (nilbuild) [inline cmd/add-member.cmd.ts]: Same here, `writeFileSync`

**Review by nilbuild — CHANGES_REQUESTED** [review]
Thanks @abidhkm just few minor comments :)

**Comment by abidhkm** [comment]
sure, addressing the comments.

**Comment by abidhkm** [comment]
comments are addressed, please review.

**Review by nilbuild — APPROVED** [review]
Thanks @abidhkm

**Pull request merged**